### PR TITLE
categories and sort-by functionality added

### PR DESCRIPTION
**Pull request by tashkenna**

### CODE REVIEW

**Review by yalicee — CHANGES_REQUESTED** [review]
Your code should have function to fetch topics from api in order to filter articles instead you are filtering articles state which could lead to stale data. 

Please reuse previously implemented get all articles function to filter by topic (using axios params).

Have a think about what you need to do for sorting as well.

File: src/components/Utils/sortArticles.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      {return articles.sort((a, b) => a.comment_count - b.comment_count);}
+
+      if (sortingOption === "comment count descending")
+      {return articles.sort((a, b) => b.comment_count - a.comment_count);}
+
+      if (sortingOption === "date ascending")
+      {return articles.sort((a, b) =>  new Date(a.created_at) - new Date(b.created_at))}
+  
+      return articles.sort((a, b) =>  new Date(b.created_at) - new Date(a.created_at))
+  };
```
Comment (robert112233) [inline src/components/Utils/sortArticles.js:18]: You shouldn't need to use this logic as it already exists on the backend!

**Review by robert112233 — COMMENTED** [review]
It looks like you are still not making use of the backends ability to filter and sort. For example:

api/articles?topic=coding

**Review by MKD01 — APPROVED** [review]
Great work, im happy for you to merge this in 👍

**Pull request merged**